### PR TITLE
feat(common): add VisMut

### PR DESCRIPTION
**Pull request by st1page**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

a part of https://github.com/risingwavelabs/risingwave/pull/11070, which wants to modify the Vis in place and avoid clone. And be safe to forbid change the length

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/common/src/buffer/bitmap.rs [hunk 1]
```diff
@@ -69,6 +86,14 @@ impl BitmapBuilder {
         }
     }
 
+    /// Creates a new bitmap with all bits set to 1.
+    pub fn filled(len: usize) -> BitmapBuilder {
+        BitmapBuilder {
+            len,
+            data: vec![usize::MAX; Bitmap::vec_len(len)],
```
Comment (wangrunji0408) [inline src/common/src/buffer/bitmap.rs:93]: Should set tailing bits to zero.

**Review by wangrunji0408 — COMMENTED** [review]
What is this for? 👀

Actually I was planning to replace all `Vis` by `Bitmap` after #9329.

**Comment by st1page** [comment]
> What is this for? 👀
> 
> Actually I was planning to replace all `Vis` by `Bitmap` after #9329.

It is a part of https://github.com/risingwavelabs/risingwave/pull/11070, which wants to modify the Vis in place and avoid clone

File: src/common/src/buffer/bitmap.rs (hunk 1 above)
Comment (st1page) [reply src/common/src/buffer/bitmap.rs]: fixed

File: src/common/src/array/vis.rs [hunk 2]
```diff
@@ -324,3 +341,179 @@ impl<'a> std::ops::Not for VisRef<'a> {
...
+            VisMutState::Builder(b) => b.is_set(idx),
+            VisMutState::Undefined => unreachable!(),
+        }
+    }
+
+    pub fn set(&mut self, n: usize, val: bool) {
+        if let VisMutState::Builder(b) = &mut self.state {
+            b.set(n, val);
+        } else {
+            let state = mem::replace(&mut self.state, VisMutState::Undefined);
```
Comment (wangrunji0408) [inline src/common/src/array/vis.rs:406]: Can we use `Compact(0)` as the placeholder so that we don't need `Undefined`?

**Review by wangrunji0408 — APPROVED** [review]
LGTM!

**Pull request merged**